### PR TITLE
Add android sample

### PR DESCRIPTION
**Pull request by raniejade**
Copied over from https://github.com/spekframework/spek-android-example

### CODE REVIEW

File: samples/android/app/build.gradle [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    implementation"org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
+    implementation 'com.android.support:appcompat-v7:27.1.1'
+    implementation 'com.android.support.constraint:constraint-layout:1.1.2'
+
+    // assertion
+    testImplementation "org.jetbrains.kotlin:kotlin-test:$kotlin_version"
+
+    // spek2
+    testImplementation "org.spekframework.spek2:spek-dsl-jvm:$spek_version"
+    // testRuntimeOnly not working, todo: fix plugin
```
Comment (vanniktech) [inline samples/android/app/build.gradle:63]: insert an issue or WIP pr here so people can watch the progress?

File: samples/android/app/build.gradle [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+// junit5 doesn't support android projects out of the box
+apply plugin: "de.mannodermaus.android-junit5"
+
+android {
+    compileSdkVersion 27
+    defaultConfig {
+        applicationId "org.spekframework.speksample"
+        minSdkVersion 24
+        targetSdkVersion 27
```
Comment (vanniktech) [inline samples/android/app/build.gradle:14]: we can probably target 28

File: samples/android/app/proguard-rules.pro [hunk 3]
```diff
@@ -0,0 +1,21 @@
+# Add project specific ProGuard rules here.
+# You can control the set of applied configuration files using the
+# proguardFiles setting in build.gradle.
+#
+# For more details, see
+#   http://developer.android.com/guide/developing/tools/proguard.html
+
+# If your project uses WebView with JS, uncomment the following
+# and specify the fully qualified class name to the JavaScript interface
+# class:
```
Comment (vanniktech) [inline samples/android/app/proguard-rules.pro]: We can empty out that file. Those comments don't add any value

File: samples/android/app/src/androidTest/java/org/spekframework/speksample/ExampleInstrumentedTest.kt [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+import android.support.test.InstrumentationRegistry
+import org.junit.Assert.assertEquals
+import org.junit.Test
+
+/**
+ * Instrumented test, which will execute on an Android device.
+ *
+ * See [testing documentation](http://d.android.com/tools/testing).
+ */
+class ExampleInstrumentedTest {
```
Comment (vanniktech) [inline samples/android/app/src/androidTest/java/org/spekframework/speksample/ExampleInstrumentedTest.kt]: Should we keep this one? It just distracts everyone

File: samples/android/app/src/main/kotlin/org/spekframework/speksample/Calculator.kt [hunk 5]
```diff
@@ -0,0 +1,6 @@
+package org.spekframework.speksample
+
+
```
Comment (vanniktech) [inline samples/android/app/src/main/kotlin/org/spekframework/speksample/Calculator.kt]: nit: double spacing

File: samples/android/build.gradle [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+    repositories {
+        google()
+        jcenter()
+    }
+    dependencies {
+        classpath 'com.android.tools.build:gradle:3.1.4'
+        classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
+        classpath "de.mannodermaus.gradle.plugins:android-junit5:$android_junit5_version"
+        classpath "org.jacoco:org.jacoco.core:$jacoco_version"
+        // NOTE: Do not place your application dependencies here; they belong
```
Comment (vanniktech) [inline samples/android/build.gradle:14]: We can get rid of this

File: samples/android/build.gradle [hunk 7]
```diff
@@ -0,0 +1,29 @@
+// Top-level build file where you can add configuration options common to all sub-projects/modules.
+
+buildscript {
+    ext.kotlin_version = '1.2.60'
```
Comment (vanniktech) [inline samples/android/build.gradle:4]: 1.2.61

File: samples/android/build.gradle [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+
+allprojects {
+    repositories {
+        google()
+        jcenter()
+        maven { url "https://dl.bintray.com/spekframework/spek-dev" }
+    }
+}
+
+task clean(type: Delete) {
```
Comment (vanniktech) [inline samples/android/build.gradle]: I don't think this is needed anymore

File: samples/android/build.gradle [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+        classpath "org.jacoco:org.jacoco.core:$jacoco_version"
+        // NOTE: Do not place your application dependencies here; they belong
+        // in the individual module build.gradle files
+    }
+}
+
+allprojects {
+    repositories {
+        google()
+        jcenter()
```
Comment (vanniktech) [inline samples/android/build.gradle]: We should also add `mavenCentral()` before cause sometimes jcenter is flaky

File: samples/android/non-android-lib/src/main/kotlin/org/spekframework/speksample/Hello.kt [hunk 10]
```diff
@@ -0,0 +1,6 @@
+package org.spekframework.speksample
+
```
Comment (vanniktech) [inline samples/android/non-android-lib/src/main/kotlin/org/spekframework/speksample/Hello.kt:2]: nit: double spacing

File: samples/android/non-android-lib/build.gradle [hunk 11]
```diff
@@ -0,0 +1,19 @@
+apply plugin: 'kotlin'
```
Comment (vanniktech) [inline samples/android/non-android-lib/build.gradle:1]: why do we put this into `samples/android/non-android` instead of `samples/kotlin`?

**Review by vanniktech — APPROVED** [review]
Can't we reuse the gradle infrastructure from the other projects instead of having multiple Gradle wrappers?

**Review by sergioboi — COMMENTED** [review]
Do we need update "Setting up" section, with correct dependencies for android projects?

File: samples/android/build.gradle (hunk 7 above)
Comment (raniejade) [reply samples/android/build.gradle:4]: I'd keep it for now, will update once I get `1.x` and `2.x` branches updated.

File: samples/android/build.gradle (hunk 8 above)
Comment (raniejade) [reply samples/android/build.gradle]: 👌

File: samples/android/non-android-lib/build.gradle (hunk 11 above)
Comment (raniejade) [reply samples/android/non-android-lib/build.gradle:1]: Still related to android, this shows that spek (ide plugin) can still be used in a non-android library within an android project.

**Pull request merged**